packages/lodestar/test/perf/api/impl/validator/attester.test.ts

```diff
@@ -0,0 +1,98 @@
+const syncStub = server.syncStub;
+chainStub.clock = server.sandbox.createStubInstance(LocalClock);
+chainStub.forkChoice = server.sandbox.createStubInstance(ForkChoice);
+chainStub.getCanonicalBlockAtSlot.resolves(ssz.phase0.SignedBeaconBlock.defaultValue());
+const dbStub = server.dbStub;
+
+const numValidators = 200000;
+const numAttachedValidators = 200;
+
+const validators = generateValidators(numValidators, {
+  effectiveBalance: MAX_EFFECTIVE_BALANCE,
+  activationEpoch: 0,
+  exitEpoch: FAR_FUTURE_EPOCH,
+});
+syncStub.isSynced.returns(true);
+server.sandbox.stub(chainStub.clock, "currentEpoch").get(() => 0);
+server.sandbox.stub(chainStub.clock, "currentSlot").get(() => 0);
+dbStub.block.get.resolves({message: {stateRoot: Buffer.alloc(32)}} as any);
+const _state = generateState(
+  {
+    slot: 0,
+    validators,
+    balances: generateInitialMaxBalances(config, numValidators),
+  },
+  config
+);
+const state = createCachedBeaconState(config, _state);
+chainStub.getHeadStateAtCurrentEpoch.resolves(state);
+
+const indices = Array.from({length: numAttachedValidators}, (_, i) => i * 5);
+
+describe("getCommitteeAssignments vs assembleAttesterDuties performance test", async () => {
+  setBenchOpts({
+    maxMs: 10 * 1000,
+    runs: 10,
+  });
+
+  // the new way of getting attester duties
+  let newDuties: AttesterDuty[] = [];
+  itBench("getCommitteeAssignments", () => {
+    const validators = state.validators.persistent;
+    const validatorData: BLSPubkey[] = [];
+    indices.forEach((index) => {
+      const validator = validators.get(index);
+      if (!validator) {
+        throw new Error(`Validator at index ${index} not in state`);
+      }
+      validatorData[index] = validator.pubkey;
+    });
+    newDuties = state.epochCtx.getCommitteeAssignments(0, validatorData);
+  });
+
+  itBench("new way (plus index2pubkey): getCommitteeAssignments", () => {
+    const validatorData: BLSPubkey[] = [];
+    indices.forEach((index) => {
+      const pubkey = state.index2pubkey[index];
+      if (!pubkey) {
+        throw new Error(`Validator pubkey at validator index ${index} not found in state.`);
+      }
+      validatorData[index] = pubkey.toBytes();
+    });
+    newDuties = state.epochCtx.getCommitteeAssignments(0, validatorData);
+  });
+
+  // the old way of getting the attester duties
+  let oldDuties: AttesterDuty[] = [];
+  itBench("old way: assembleAttesterDuty batch", () => {
+    oldDuties = [];
+    for (const validatorIndex of indices) {
+      const validator = state.validators[validatorIndex];
+      const duty = assembleAttesterDuty(config, {pubkey: validator.pubkey, index: validatorIndex}, state.epochCtx, 0);
+      if (duty) oldDuties.push(duty);
+    }
+  });
+
+  // verify that both the old and new ways get the same data
+  newDuties.sort((x, y) => x.validatorIndex - y.validatorIndex);
+  oldDuties.sort((x, y) => x.validatorIndex - y.validatorIndex);
+  expect(newDuties).to.deep.equal(oldDuties);
```

Comment:
Doing an expect() in a describe() block will kinda break mocha. expect() must only be called inside an it() block.

Tho, separate testing and benchmark in different runs. Here exclusively test performance and then assert correctnes in `test/unit/**` with a different file

Reply:
I think once we decide that the new strategy is sound, we'll just remove the oldDuties anyways, so there will be no need to compare anymore.  So we can just remove that section once the main part of this PR is complete.